addons/stock/models/stock_move.py

```diff
@@ -817,6 +821,10 @@ def _merge_moves(self, merge_into=False):
         into another existing one, return this one and not the (now unlinked) original.
         """
         distinct_fields = self._prepare_merge_moves_distinct_fields()
+        neg_qty_moves = self.filtered(lambda m: float_compare(m.product_qty, 0, precision_rounding=m.product_uom.rounding) < 0)
+        if neg_qty_moves:
+            excluded_fields = self._prepare_merge_negative_moves_excluded_distinct_fields()
```

Comment:
It won't work correctly: 
If you have 3 sibling moves: all the same excepted the created_production_id, one is negative qty (with created_production_id = False), the 2 others have different created_production_id. the 3 moves will be merged which isn't correct (the 2 with created_production_id shouldn't be merge).
IMO we should create a custom groupby for this case :/